### ISSUE TITLE
Bug: `__proto__` not registered in `/* exported */`

### ISSUE THREAD

**Issue by swwind** [eslint/eslint#21250]
### Environment

Node version: v26.3.0
npm version: v11.16.0
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: linux 7.0.0-29-generic

### What parser are you using?

Default (Espree)

### What did you do?

```js
/*eslint no-implicit-globals: "error"*/

var foo = 233;   // reports

/* exported bar */
var bar = 233;    // not reports

/* exported __proto__ */
var __proto__ = 233;   // reports
```

### What did you expect to happen?

`var __proto__ = 233;` should not reports

### What actually happened?

omg, it reports.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8taW1wbGljaXQtZ2xvYmFsczogXCJlcnJvclwiKi9cblxudmFyIGZvbyA9IDIzMztcblxuLyogZXhwb3J0ZWQgYmFyICovXG52YXIgYmFyID0gMjMzO1xuXG4vKiBleHBvcnRlZCBfX3Byb3RvX18gKi9cbnZhciBfX3Byb3RvX18gPSAyMzM7XG4iLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLWltcGxpY2l0LWdsb2JhbHMiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQifX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

I think there might be somewhere assigned `__proto__` to `{}`.

**Comment by sethamus** [comment]
I can reproduce this. I will work on it if accepted.

**Comment by DMartens** [comment]
Thank you for reporting this. I am able to reproduce this behavior.
@sethamus feel free to open a PR for this.
